BraveRewardsUI/Tipping/TippingViewController.swift

```diff
@@ -72,9 +155,33 @@ class TippingViewController: UIViewController, UIViewControllerTransitioningDele
   }
   
   @objc private func tappedSendTip() {
-    tippingView.setTippingConfirmationVisible(true, animated: true)
-    DispatchQueue.main.asyncAfter(deadline: .now() + 2.0) { [weak self] in
-      self?.dismiss(animated: true)
+    if let selectedIndex = self.tippingView.optionSelectionView.selectedOptionIndex {
+      let amount = self.tippingView.optionSelectionView.options[selectedIndex].value.doubleValue
+      
+      state.ledger.listRecurringTips { [weak self] info in
+        guard let self = self else { return }
+        
+        // If the user has recurring tips, remove the monthly tip
+        if info.firstIndex(where: { $0.id == self.publisherInfo.id }) != nil {
+          self.state.ledger.removeRecurringTip(publisherId: self.publisherInfo.id)
+        }
+        
+        // Add recurring tips if there is none..
+        if self.tippingView.optionSelectionView.isMonthly {
+          self.state.ledger.addRecurringTip(publisherId: self.publisherInfo.id, amount: amount)
+        }
+        
+        // TODO: Figure out -- Not sure why amount is of type integer..
+        // Also not sure user's currency
+        self.state.ledger.tipPublisherDirectly(self.publisherInfo, amount: Int32(amount), currency: "BAT")
+        
+        self.tippingView.setInfo(name: self.publisherInfo.name, tipAmount: amount)
```

Comment:
This name property does not include suffix string like "on Youtube" etc. to get it please use:
`let provider = " \(publisher.provider.isEmpty ? "" : String(format: Strings.OnProviderText, publisher.provider))"`

and concat this to name.